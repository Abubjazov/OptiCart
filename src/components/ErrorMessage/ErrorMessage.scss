@import '../../variables';

.error-message {
	margin-top: 30px;

	p {
		color: $deep-main;
		font-size: 90px;
		font-weight: 900;
		margin-bottom: 10px;
	}
}
