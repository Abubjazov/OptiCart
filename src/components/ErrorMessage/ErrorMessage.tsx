import { NavLink } from 'react-router-dom'

import './ErrorMessage.scss'

export const ErrorMessage = ({ error }: ErrorMessageProps): JSX.Element => (
	<div className='error-message'>
		<p>{error}</p>

		<NavLink to='/optiCart/'>Back to main page</NavLink>
	</div>
)

export interface ErrorMessageProps {
	error: string | null
}
