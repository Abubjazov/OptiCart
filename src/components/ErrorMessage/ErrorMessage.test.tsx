import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { ErrorMessage } from './ErrorMessage'

describe('Component: ErrorMessage', () => {
	test('should render ErrorMessage', () => {
		const { asFragment } = render(
			<BrowserRouter>
				<ErrorMessage error={'Test Error String'} />
			</BrowserRouter>
		)

		expect(asFragment()).toMatchSnapshot()
	})

	// test('should navigatr home page', () => {
	// 	render(
	// 		<BrowserRouter>
	// 			<ErrorMessage error={'Test Error String'} />
	// 		</BrowserRouter>
	// 	)

	// 	const backButton = screen.getByText(/Back to main page/)

	// 	userEvent.click(backButton)
	// 	const linkElement = screen.logTestingPlaygroundURL()
	// 	expect(linkElement).toBeInTheDocument()
	// })
})
