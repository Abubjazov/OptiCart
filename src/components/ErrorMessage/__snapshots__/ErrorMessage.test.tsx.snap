// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ErrorMessage should render ErrorMessage 1`] = `
<DocumentFragment>
  <div
    class="error-message"
  >
    <p
      style="text-align: center; font-weight: 300; font-size: 24px;"
    >
      Test Error String
    </p>
    <a
      aria-current="page"
      class="active"
      href="/"
      style="display: block; text-align: center; font-weight: 300; font-size: 24px; margin-top: 13px; color: rgb(66, 145, 136);"
    >
      Back to main page
    </a>
  </div>
</DocumentFragment>
`;

exports[`renders ErrorMessage 1`] = `
<DocumentFragment>
  <div
    class="error-message"
  >
    <p
      style="text-align: center; font-weight: 300; font-size: 24px;"
    >
      Test Error String
    </p>
    <a
      aria-current="page"
      class="active"
      href="/"
      style="display: block; text-align: center; font-weight: 300; font-size: 24px; margin-top: 13px; color: rgb(66, 145, 136);"
    >
      Back to main page
    </a>
  </div>
</DocumentFragment>
`;
